bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitSecuritySystemImpl.java

```diff
@@ -0,0 +1,143 @@
+import io.github.hapjava.characteristics.impl.securitysystem.TargetSecuritySystemStateEnum;
+import io.github.hapjava.services.impl.SecuritySystemService;
+
+/**
+ *  Implements SecuritySystem as a GroupedAccessory made up of multiple items:
+ *  <ul>
+ *  <li>CurrentSecuritySystemState: String type</li>
+ *  <li>TargetSecuritySystemState: String type</li>
+ *  </ul>
+ * @author Cody Cutrer - Initial contribution
+ */
+public class HomekitSecuritySystemImpl extends AbstractHomekitAccessoryImpl implements SecuritySystemAccessory {
+    private static final Logger LOGGER = LoggerFactory.getLogger(HomekitSecuritySystemImpl.class);
+
+    public HomekitSecuritySystemImpl(HomekitTaggedItem taggedItem, List<HomekitTaggedItem> mandatoryCharacteristics, HomekitAccessoryUpdater updater, HomekitSettings settings) throws IncompleteAccessoryException {
+        super(taggedItem, mandatoryCharacteristics, updater, settings);
+        getServices().add(new SecuritySystemService(this));
+    }
+
+    @Override
+    public CompletableFuture<CurrentSecuritySystemStateEnum> getCurrentSecuritySystemState() {
+        CurrentSecuritySystemStateEnum state;
+        final @Nullable State itemState = getStateAs(HomekitCharacteristicType.SECURITY_SYSTEM_CURRENT_STATE, StringType.class);
+        if (itemState!=null) {
+            String stringValue = itemState.toString();
+            if (stringValue.equalsIgnoreCase("DISARMED")) {
+                state = CurrentSecuritySystemStateEnum.DISARMED;
+            } else if (stringValue.equalsIgnoreCase("AWAY_ARM")) {
+                state = CurrentSecuritySystemStateEnum.AWAY_ARM;
+            } else if (stringValue.equalsIgnoreCase("STAY_ARM")) {
+                state = CurrentSecuritySystemStateEnum.STAY_ARM;
+            } else if (stringValue.equalsIgnoreCase("NIGHT_ARM")) {
+                state = CurrentSecuritySystemStateEnum.NIGHT_ARM;
+            } else if (stringValue.equalsIgnoreCase("TRIGGERED")) {
+                state = CurrentSecuritySystemStateEnum.TRIGGERED;
+            } else if (stringValue.equals("UNDEF") || stringValue.equals("NULL")) {
+                LOGGER.debug("Security system target state not available. Relaying value of DISARM to HomeKit");
+                state = CurrentSecuritySystemStateEnum.DISARMED;
+            } else {
+                LOGGER.error(
+                    "Unrecognized security system target state: {}. Expected DISARM, AWAY_ARM, STAY_ARM, NIGHT_ARM strings in value.",
+                    stringValue);
+                state = CurrentSecuritySystemStateEnum.DISARMED;
+            }
+        } else {
+            LOGGER.debug("Security system target state not available. Relaying value of DISARM to HomeKit");
+            state = CurrentSecuritySystemStateEnum.DISARMED;
+        }
+        return CompletableFuture.completedFuture(state);
+    }
+    @Override
+    public void setTargetSecuritySystemState(final TargetSecuritySystemStateEnum state) throws Exception {
```

Comment:
Why is this method throwing `Exception`?

Reply:
yep. removed